build/ci/phase-template.yml

```diff
@@ -0,0 +1,32 @@
+parameters:
+  name: ''
+  buildScript: ''
+  queue: ''
+
+phases:
+  - phase: ${{ parameters.name }}
+    variables:
+      _buildScript: ${{ parameters.buildScript }}
+      _phaseName: ${{ parameters.name }}
+    queue:
+      parallel: 2
+      matrix:
+        Build_Debug:
+          _configuration: Debug
+        Build_Release:
+          _configuration: Release 
+      ${{ insert }}: ${{ parameters.queue }}
+    steps:
+    - script: $(_buildScript) -$(_configuration) -runtests
+      displayName: Build and Test
+    - task: PublishTestResults@2
+      displayName: Publish Test Results
+      condition: succeededOrFailed()
+      inputs:
+        testRunner: 'vSTest'
```

Comment:
>vSTest [](start = 21, length = 6)

don't we use xUnit tests, or they interchangeable?

Reply:
We use xunit as a framework, but we use VSTest as our test runner:
https://github.com/dotnet/machinelearning/blob/master/test/run-tests.proj#L10